diff --git a/basctl/Module_basctl.mk b/basctl/Module_basctl.mk
index 92d3553..5da59c9 100644
--- a/basctl/Module_basctl.mk
+++ b/basctl/Module_basctl.mk
@@ -31,7 +31,7 @@ $(eval $(call gb_Module_add_l10n_targets,basctl,\
 
 endif
 
+$(eval $(call gb_Module_add_l10n_targets,basctl,\
-$(eval $(call gb_Module_add_targets,basctl,\
 	UIConfig_basicide \
 ))
 
diff --git a/chart2/Module_chart2.mk b/chart2/Module_chart2.mk
index a7751e9..49a8c51 100644
--- a/chart2/Module_chart2.mk
+++ b/chart2/Module_chart2.mk
@@ -12,11 +12,11 @@ $(eval $(call gb_Module_Module,chart2))
 $(eval $(call gb_Module_add_targets,chart2,\
     Library_chartcontroller \
     Library_chartcore \
-	UIConfig_chart2 \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,chart2,\
     AllLangResTarget_chartcontroller \
+	UIConfig_chart2 \
 ))
 
 $(eval $(call gb_Module_add_slowcheck_targets,chart2,\
diff --git a/cui/Module_cui.mk b/cui/Module_cui.mk
index 2a2fca4..62398b3 100644
--- a/cui/Module_cui.mk
+++ b/cui/Module_cui.mk
@@ -11,11 +11,11 @@ $(eval $(call gb_Module_Module,cui))
 
 $(eval $(call gb_Module_add_targets,cui,\
 	Library_cui \
-	UIConfig_cui \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,cui,\
 	AllLangResTarget_cui \
+	UIConfig_cui \
 ))
 
 # vim: set noet sw=4 ts=4:
diff --git a/dbaccess/Module_dbaccess.mk b/dbaccess/Module_dbaccess.mk
index b9d7d56..5c05968 100644
--- a/dbaccess/Module_dbaccess.mk
+++ b/dbaccess/Module_dbaccess.mk
@@ -18,6 +18,13 @@ $(eval $(call gb_Module_add_targets,dbaccess,\
 	Library_dbmm \
 	Library_dbu \
 	Library_sdbt \
+))
+
+$(eval $(call gb_Module_add_l10n_targets,dbaccess,\
+    AllLangResTarget_dba \
+    AllLangResTarget_dbmm \
+    AllLangResTarget_dbu \
+    AllLangResTarget_sdbt \
 	UIConfig_dbaccess \
 	UIConfig_dbapp \
 	UIConfig_dbbrowser \
@@ -27,13 +34,6 @@ $(eval $(call gb_Module_add_l10n_targets,dbaccess,\
 	UIConfig_dbtdata \
 ))
 
-$(eval $(call gb_Module_add_l10n_targets,dbaccess,\
-    AllLangResTarget_dba \
-    AllLangResTarget_dbmm \
-    AllLangResTarget_dbu \
-    AllLangResTarget_sdbt \
-))
-
 ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
 $(eval $(call gb_Module_add_check_targets,dbaccess,\
     CppunitTest_dbaccess_firebird_test \
diff --git a/desktop/Module_desktop.mk b/desktop/Module_desktop.mk
index ed9e4d9..c90de67 100644
--- a/desktop/Module_desktop.mk
+++ b/desktop/Module_desktop.mk
@@ -20,13 +20,13 @@ $(eval $(call gb_Module_add_targets,desktop,\
     $(if $(ENABLE_HEADLESS),,Library_spl) \
     Package_branding \
     $(if $(CUSTOM_BRAND_DIR),Package_branding_custom) \
-    UIConfig_deployment \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,desktop,\
     AllLangResTarget_deployment \
     AllLangResTarget_deploymentgui \
     AllLangResTarget_dkt \
+    UIConfig_deployment \
 ))
 
 ifeq ($(OS),LINUX)
diff --git a/extensions/Module_extensions.mk b/extensions/Module_extensions.mk
index 4761bba..4f30191 100644
--- a/extensions/Module_extensions.mk
+++ b/extensions/Module_extensions.mk
@@ -34,13 +34,13 @@ $(eval $(call gb_Module_add_targets,extensions,\
 	Library_bib \
 	Library_dbp \
 	Library_pcr \
-	UIConfig_sbibliography \
-	UIConfig_spropctrlr \
 ))
 $(eval $(call gb_Module_add_l10n_targets,extensions,\
 	AllLangResTarget_bib \
 	AllLangResTarget_dbp \
 	AllLangResTarget_pcr \
+	UIConfig_sbibliography \
+	UIConfig_spropctrlr \
 ))
 endif
 
diff --git a/filter/Module_filter.mk b/filter/Module_filter.mk
index 403184a..acff4f8 100644
--- a/filter/Module_filter.mk
+++ b/filter/Module_filter.mk
@@ -61,7 +61,6 @@ $(eval $(call gb_Module_add_targets,filter,\
 	Package_docbook \
 	Package_xhtml \
 	Package_xslt \
-	UIConfig_xsltdlg \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,filter,\
@@ -69,6 +68,7 @@ $(eval $(call gb_Module_add_l10n_targets,filter,\
 	AllLangResTarget_pdffilter \
 	AllLangResTarget_t602filter \
 	AllLangResTarget_xsltdlg \
+	UIConfig_xsltdlg \
 ))
 
 ifneq (,$(filter DESKTOP,$(BUILD_TYPE)))
diff --git a/framework/Module_framework.mk b/framework/Module_framework.mk
index 2e6d5b7..b597d34 100644
--- a/framework/Module_framework.mk
+++ b/framework/Module_framework.mk
@@ -26,11 +26,11 @@ $(eval $(call gb_Module_add_targets,framework,\
     Library_fwl \
     Library_fwm \
     Package_dtd \
-    UIConfig_startmodule \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,framework,\
     AllLangResTarget_fwe \
+    UIConfig_startmodule \
 ))
 
 ifneq ($(OOO_JUNIT_JAR),)
diff --git a/padmin/Module_padmin.mk b/padmin/Module_padmin.mk
index 45db7aa..12f3f4c 100644
--- a/padmin/Module_padmin.mk
+++ b/padmin/Module_padmin.mk
@@ -26,11 +26,11 @@ $(eval $(call gb_Module_add_targets,padmin,\
     Executable_spadmin.bin \
     Library_spa \
     Package_padmin \
-    UIConfig_spa \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,padmin,\
     AllLangResTarget_spa \
+    UIConfig_spa \
 ))
 
 endif
diff --git a/reportdesign/Module_reportdesign.mk b/reportdesign/Module_reportdesign.mk
index bba86b9..620fbbf 100644
--- a/reportdesign/Module_reportdesign.mk
+++ b/reportdesign/Module_reportdesign.mk
@@ -13,12 +13,12 @@ $(eval $(call gb_Module_add_targets,reportdesign,\
     Library_rpt \
     Library_rptui \
     Library_rptxml \
-    UIConfig_dbreport \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,reportdesign,\
     AllLangResTarget_rpt \
     AllLangResTarget_rptui \
+    UIConfig_dbreport \
 ))
 
 # deactivated since sb123;
diff --git a/sc/Module_sc.mk b/sc/Module_sc.mk
index 6730163..cc974a0 100644
--- a/sc/Module_sc.mk
+++ b/sc/Module_sc.mk
@@ -15,11 +15,11 @@ $(eval $(call gb_Module_add_targets,sc,\
 	Library_scd \
 	Library_scfilt \
 	Library_scui \
-	UIConfig_scalc \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,sc,\
 	AllLangResTarget_sc \
+	UIConfig_scalc \
 ))
 
 ifneq (,$(ENABLE_OPENCL))
diff --git a/sd/Module_sd.mk b/sd/Module_sd.mk
index 3321279..16e7772 100644
--- a/sd/Module_sd.mk
+++ b/sd/Module_sd.mk
@@ -17,12 +17,12 @@ $(eval $(call gb_Module_add_targets,sd,\
     Package_opengl \
     Package_web \
     Package_xml \
-    UIConfig_sdraw \
-    UIConfig_simpress \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,sd,\
     AllLangResTarget_sd \
+    UIConfig_sdraw \
+    UIConfig_simpress \
 ))
 
 ifneq ($(OS),DRAGONFLY)
diff --git a/sfx2/Module_sfx2.mk b/sfx2/Module_sfx2.mk
index be82c7c..1164ca4 100644
--- a/sfx2/Module_sfx2.mk
+++ b/sfx2/Module_sfx2.mk
@@ -21,11 +21,11 @@ $(eval $(call gb_Module_Module,sfx2))
 
 $(eval $(call gb_Module_add_targets,sfx2,\
     Library_sfx \
-    UIConfig_sfx \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,sfx2,\
     AllLangResTarget_sfx2 \
+    UIConfig_sfx \
 ))
 
 $(eval $(call gb_Module_add_check_targets,sfx2,\
diff --git a/solenv/gbuild/Module.mk b/solenv/gbuild/Module.mk
index ee1cb46..dd35658 100644
--- a/solenv/gbuild/Module.mk
+++ b/solenv/gbuild/Module.mk
@@ -227,7 +227,7 @@ endif
 endef
 
 define gb_Module_add_target
+$(if $(filter AllLang% Dictionary% UIConfig%,$(2)),$(warning target $(2) should be a l10n target))
-$(if $(filter AllLang% Dictionary%,$(2)),$(warning target $(2) should be a l10n target))
 $(call gb_Module__read_targetfile,$(1),$(2),target)
 
 $(call gb_Module_get_nonl10n_target,$(1)) : $$(gb_Module_CURRENTTARGET)
@@ -236,7 +236,7 @@ $(call gb_Module_get_clean_target,$(1)) : $$(gb_Module_CURRENTCLEANTARGET)
 endef
 
 define gb_Module_add_l10n_target
+$(if $(filter AllLang% Dictionary% UIConfig%,$(2)),,$(warning target $(2) should not be a l10n target))
-$(if $(filter AllLang% Dictionary%,$(2)),,$(warning target $(2) should not be a l10n target))
 $(call gb_Module__read_targetfile,$(1),$(2),target)
 
 $(call gb_Module_get_l10n_target,$(1)) : $$(gb_Module_CURRENTTARGET)
diff --git a/starmath/Module_starmath.mk b/starmath/Module_starmath.mk
index 2830a46..bd434dd 100644
--- a/starmath/Module_starmath.mk
+++ b/starmath/Module_starmath.mk
@@ -13,11 +13,11 @@ $(eval $(call gb_Module_Module,starmath))
 $(eval $(call gb_Module_add_targets,starmath,\
     Library_sm \
     Library_smd \
-    UIConfig_smath \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,starmath,\
     AllLangResTarget_sm \
+    UIConfig_smath \
 ))
 
 $(eval $(call gb_Module_add_check_targets,starmath,\
diff --git a/svtools/Module_svtools.mk b/svtools/Module_svtools.mk
index f96a758..e2a3512 100644
--- a/svtools/Module_svtools.mk
+++ b/svtools/Module_svtools.mk
@@ -22,11 +22,11 @@ $(eval $(call gb_Module_Module,svtools))
 $(eval $(call gb_Module_add_targets,svtools,\
     Library_hatchwindowfactory \
     Library_svt \
-    UIConfig_svt \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,svtools,\
     AllLangResTarget_svt \
+    UIConfig_svt \
 ))
 
 ifneq ($(CROSS_COMPILING),YES)
diff --git a/svx/Module_svx.mk b/svx/Module_svx.mk
index 80f2bd7..071e36f 100644
--- a/svx/Module_svx.mk
+++ b/svx/Module_svx.mk
@@ -23,6 +23,7 @@ $(eval $(call gb_Module_add_targets,svx,\
     Library_svx \
     Library_svxcore \
     Library_textconversiondlgs \
-    UIConfig_svx \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,svx,\
@@ -30,7 +31,6 @@ $(eval $(call gb_Module_add_l10n_targets,svx,\
     AllLangResTarget_ofa \
     AllLangResTarget_gal \
     AllLangResTarget_textconversiondlgs \
+    UIConfig_svx \
 ))
 
 ifneq (,$(filter DESKTOP,$(BUILD_TYPE)))
diff --git a/sw/Module_sw.mk b/sw/Module_sw.mk
index d5d3f44..2e35087 100644
--- a/sw/Module_sw.mk
+++ b/sw/Module_sw.mk
@@ -26,6 +26,10 @@ $(eval $(call gb_Module_add_targets,sw,\
 	Library_sw \
 	Library_swd \
 	Library_swui \
+))
+
+$(eval $(call gb_Module_add_l10n_targets,sw,\
+    AllLangResTarget_sw \
 	UIConfig_sglobal \
 	UIConfig_sweb \
 	UIConfig_swform \
@@ -35,10 +39,6 @@ $(eval $(call gb_Module_add_l10n_targets,sw,\
 	UIConfig_qa \
 ))
 
-$(eval $(call gb_Module_add_l10n_targets,sw,\
-    AllLangResTarget_sw \
-))
-
 ifneq ($(DISABLE_SCRIPTING),TRUE)
 
 $(eval $(call gb_Module_add_targets,sw,\
diff --git a/uui/Module_uui.mk b/uui/Module_uui.mk
index 193bea5..c827e41 100644
--- a/uui/Module_uui.mk
+++ b/uui/Module_uui.mk
@@ -11,11 +11,11 @@ $(eval $(call gb_Module_Module,uui))
 
 $(eval $(call gb_Module_add_targets,uui,\
 	Library_uui \
-	UIConfig_uui \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,uui,\
 	AllLangResTarget_uui \
+	UIConfig_uui \
 ))
 
 # vim: set noet sw=4 ts=4:
diff --git a/vcl/Module_vcl.mk b/vcl/Module_vcl.mk
index 4dc6434..a65736d 100644
--- a/vcl/Module_vcl.mk
+++ b/vcl/Module_vcl.mk
@@ -25,11 +25,11 @@ $(eval $(call gb_Module_add_targets,vcl,\
     $(if $(filter DESKTOP,$(BUILD_TYPE)), \
 		StaticLibrary_vclmain \
 		Executable_ui-previewer) \
-    UIConfig_vcl \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,vcl,\
     AllLangResTarget_vcl \
+    UIConfig_vcl \
 ))
 
 ifeq ($(GUIBASE),unx)
diff --git a/xmlsecurity/Module_xmlsecurity.mk b/xmlsecurity/Module_xmlsecurity.mk
index e88859d..d29444c 100644
--- a/xmlsecurity/Module_xmlsecurity.mk
+++ b/xmlsecurity/Module_xmlsecurity.mk
@@ -13,11 +13,11 @@ $(eval $(call gb_Module_add_targets,xmlsecurity,\
 	Library_xmlsecurity \
 	Library_xsec_fw \
 	$(if $(filter-out IOS ANDROID,$(OS)),Library_xsec_xmlsec) \
-	UIConfig_xmlsec \
 ))
 
 $(eval $(call gb_Module_add_l10n_targets,xmlsecurity,\
 	AllLangResTarget_xsec \
+	UIConfig_xmlsec \
 ))
 
 #FIXME: ^^^, get nss&libxmlsec building on ios and android
